include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -1249,3 +1249,131 @@ def EndOp : Handshake_Op<"end", [Terminator]> {
     mlir::ValueRange getMemoryControls();
   }];
 }
+
+//===----------------------------------------------------------------------===//
+// Speculation operations
+//===----------------------------------------------------------------------===//
+
+def SpeculatorOp : Handshake_Op<"speculator", [ 
+  TypesMatchWith<"data operand type matches data result type",
+                 "dataIn", "dataOut", "$_self">]> {
+  let summary = "Central control unit of the speculative circuit.";
+  let description = [{
+    The speculator produces speculative tokens when real data has
+    not yet arrived. The speculator is also responsible for deciding 
+    the correctness of its speculation.
+
+    It needs to send control tokens to save units, commit units, 
+    and save-commit units.
+
+    Example:
+    ```mlir
+    %dataOut, %saveCtrl, %commitCtrl, %SCControl1, %SCControl2, 
+      %SCBranchControl = "handshake.speculator"(%dataOut) 
+      {name = #handshake.name<"speculator0">} : 
```

Comment:
nit pick: non-intrinsic attributes (`name`) should not appear here, delete the attribute dictionary from the example. Same comment applies to other operations.